diff -u -r ../glade3ref/glade-3-2.90.0/src/glade-parser.c ./src/glade-parser.c
--- ../glade3ref/glade-3-2.90.0/src/glade-parser.c	2005-09-15 03:45:43.000000000 +0200
+++ ./src/glade-parser.c	2005-12-22 18:08:43.000000000 +0100
@@ -1143,6 +1143,24 @@
 }
 
 /**
+ * glade_interface_new
+ *
+ * Creates a GladeInterface structure.
+ */
+GladeInterface *
+glade_interface_new ()
+{
+	GladeInterface *interface;
+	interface = g_new0 (GladeInterface, 1);
+	interface->names = g_hash_table_new (g_str_hash, g_str_equal);
+	interface->strings = g_hash_table_new_full (g_str_hash,
+						    g_str_equal,
+						    (GDestroyNotify)g_free,
+						    NULL);
+	return interface;
+}
+
+/**
  * glade_interface_destroy
  * @interface: the GladeInterface structure.
  *
diff -u -r ../glade3ref/glade-3-2.90.0/src/glade-parser.h ./src/glade-parser.h
--- ../glade3ref/glade-3-2.90.0/src/glade-parser.h	2005-09-15 03:45:44.000000000 +0200
+++ ./src/glade-parser.h	2005-12-22 18:06:39.000000000 +0100
@@ -128,6 +128,7 @@
 					   const gchar *domain);
 GladeInterface *glade_parser_parse_buffer (const gchar *buffer, gint len,
 					   const gchar *domain);
+GladeInterface *glade_interface_new ();
 void            glade_interface_destroy   (GladeInterface  *interface);
 void            glade_interface_dump      (GladeInterface  *interface,
 					   const gchar     *filename);
diff -u -r ../glade3ref/glade-3-2.90.0/src/glade-widget.c ./src/glade-widget.c
--- ../glade3ref/glade-3-2.90.0/src/glade-widget.c	2005-09-29 18:25:16.000000000 +0200
+++ ./src/glade-widget.c	2005-12-14 14:50:41.000000000 +0100
@@ -1765,13 +1765,21 @@
 glade_widget_retrieve_from_position (GtkWidget *base, int x, int y)
 {
 	GtkWidget *toplevel_widget;
+	GtkWidget *next_widget;
 	gint top_x;
 	gint top_y;
 	
-	toplevel_widget = gtk_widget_get_toplevel (base);
+	next_widget = base;
+	toplevel_widget = base;
+	while (next_widget && glade_widget_get_from_gobject (next_widget)) {
+		toplevel_widget = next_widget;
+		next_widget = gtk_widget_get_parent (next_widget);
+	}
+	
+/*	toplevel_widget = gtk_widget_get_toplevel (base);
 	if (!GTK_WIDGET_TOPLEVEL (toplevel_widget))
 		return NULL;
-
+*/
 	gtk_widget_translate_coordinates (base, toplevel_widget, x, y, &top_x, &top_y);
 	return glade_widget_find_deepest_child_at_position
 		(GTK_CONTAINER (toplevel_widget),
